src/main/scala/io/qbeast/spark/delta/DefaultFileIndex.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.delta
+
+import io.qbeast.spark.index.query.QueryFiltersUtils
+import org.apache.hadoop.fs.Path
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.delta.files.TahoeLogFileIndex
+import org.apache.spark.sql.delta.DeltaLog
+import org.apache.spark.sql.execution.datasources.FileIndex
+import org.apache.spark.sql.execution.datasources.PartitionDirectory
+import org.apache.spark.sql.execution.SQLExecution
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.SparkSession
+
+/**
+ * Default implementation of the FileIndex.
+ *
+ * @param target
+ *   the target file index implemented by Delta
+ */
+class DefaultFileIndex private (target: TahoeLogFileIndex)
```

Comment:
Should we include Qbeast on the name of the class? Like "QbeastDefaultFileIndex"?

Reply:
I do not think so, here everything is Qbeast, adding Qbeast does not add any new information to what the abstraction is dedicated to. IMHO adding Qbeast only makes sense for public classes to be used by the end-user, because for the end- user is important to distinguish them from similar concepts from other libraries. For internal API we can use a shorter name.